### DISCUSSION TITLE
Logging - Begin Scope in Listener

### DISCUSSION THREAD

**Discussion by fbrum** [quartznet/quartznet#1745] [Ideas]
Sadly when using a Job Listener or a Trigger Listener, it is not possible to set a logging scope to be reused throughout the job.

Basically this is what I'm trying to do:

```C#
public Task JobToBeExecuted(IJobExecutionContext context, CancellationToken cancellationToken = new ())
    {
        var state = new Dictionary<string, object?>
        {
            { "CorrelationId", Guid.NewGuid() },
            { "JobName", context.JobDetail.Key}
        };
        _logger.BeginScope(state);
        _logger.LogInformation("Starting job at {timestamp}", context.FireTimeUtc);
        return Task.CompletedTask;
    }
```
I suppose the job is not on the same thread as the Listener, but this should be possible.
I used to have a autofac interceptor on a older version of QUARTZ, but that doesn't work anymore. But we should need to do that when there are native listeners available.
Can you guys check that?

**Comment by lahma** [comment]
This sounds more like custom `JobRunShell`, creating a decorator around `Run` probably with inheritance. There's been [discussion](https://github.com/quartznet/quartznet/discussions/1348) about having ability to create and register a custom `IJobRunShellFactory` and that would be needed for this.
